​
# Exercise

The figure below is the RF front end block diagram of the Motorola M12 Oncore GPS receiver. What are the IF signal frequencies after the first and second mixer down-conversion ($f_{IF_1}$ and $f_{IF_2}$)?

![Motorola M12 Oncore GPS Receiver RF Front End Block Diagram](./diagram.png)

In [5]:
f_L1 = 1.57542e9      # L1 Frequency
f_LO = 16.368e6       # Local Oscillator Frequency
f_M1 = f_LO * 100     # First Mixer Frequency
f_IF1 = f_L1 - f_M1   # First Intermediate Frequency
f_M2 = f_LO * 4       # Second Mixer Frequency
f_IF2 = f_IF1 + f_M2  # Second Intermediate Frequency;  note the addition here*

# The addition in the second mixer stage occurs because the mixer
# can produce both sum and difference frequencies, and in this case,
# the sum frequency is used because it is closer to 0 Hz.

print(f"First Intermediate Frequency (f_IF1): {f_IF1 / 1e6:.2f} MHz")
print(f"Second Intermediate Frequency (f_IF2): {f_IF2 / 1e6:.2f} MHz")

First Intermediate Frequency (f_IF1): -61.38 MHz
Second Intermediate Frequency (f_IF2): 4.09 MHz


First Intermediate Frequency (f_IF1): -61.38 MHz

Second Intermediate Frequency (f_IF2): 4.09 MHz
